pkg/kubelet/apis/config/types.go

```diff
@@ -382,6 +385,12 @@ type KubeletConfiguration struct {
 	// Defaults to 10 seconds, requires GracefulNodeShutdown feature gate to be enabled.
 	// For example, if ShutdownGracePeriod=30s, and ShutdownGracePeriodCriticalPods=10s, during a node shutdown the first 20 seconds would be reserved for gracefully terminating normal pods, and the last 10 seconds would be reserved for terminating critical pods.
 	ShutdownGracePeriodCriticalPods metav1.Duration
+	// A comma separated list of bracket-enclosed configurations for memory manager.
+	// Each configuration describes pre-reserved memory for the particular memory type on a specific NUMA node.
+	// The Memory Manager validates whether total amount of pre-reserved memory is identical to reserved-memory by the Node Allocatable feature.
+	// The format is {numa-node=integer, memory-type=string, limit=string}
+	// (e.g. {numa-node=0, type=memory, limit=1Gi}, {numa-node=1, type=memory, limit=1Gi})
+	ReservedMemory []map[string]string
```

Comment:
I don't think this is appropriate for our configuration files, which is a true API.  This should be an array of structs:

```
ReservedMemory []MemoryReservation
...

type MemoryReservation struct {
  NumaNode int32
  Limit v1.ResourceList
}
```

Alternatively, I suppose you could model it as 

```
// Map of numa nodes to memory limit reservations
ReservedMemory map[int]v1.ResourceList
```

But we would need to be absolutely sure we will never add additional complexity to the memory reservation (i.e. we can only ever have a resource map)

Reply:
(and the flagging as violation was appropriate, so I don't think this is approvable as is)

Reply:
changed the parameter to have the form `ReservedMemory []MemoryReservation`
done